### PR TITLE
Improve documentation on rasterization

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Inspired by, but independent of #18956.

### CODE REVIEW

File: examples/misc/rasterization_demo.py [hunk 1]
```diff
@@ -27,37 +42,34 @@
...
+# Even though the text is below the rasterization_zorder, it is not rasterized
+# because Text objects do not support rasterization.
 ax4.set_aspect(1)
 m = ax4.pcolormesh(xx, yy, d, zorder=-20)
 ax4.text(0.5, 0.5, "Text", alpha=0.2, zorder=-15,
          va="center", ha="center", size=50, transform=ax4.transAxes)
-# Set zorder value below which artists will be rasterized
 ax4.set_rasterization_zorder(-10)
 ax4.set_title("Rasterization z$<-10$")
-# ax2.title.set_rasterized(True) # should display a warning
```
Comment (anntzer) [inline examples/misc/rasterization_demo.py:60]: that's not true anymore

File: examples/misc/rasterization_demo.py [hunk 2]
```diff
@@ -27,37 +42,34 @@
...
 ax3.pcolormesh(xx, yy, d)
 ax3.text(0.5, 0.5, "Text", alpha=0.2,
          va="center", ha="center", size=50, transform=ax3.transAxes)
 ax3.set_title("No Rasterization")
 
-# Create a pseudocolor rastertized plot with a non-regular rectangular
-# grid and an overlapped "Text"
+# pcolormesh with an overlaid text without rasterization; enabled by zorder
+# Even though the text is below the rasterization_zorder, it is not rasterized
+# because Text objects do not support rasterization.
```
Comment (anntzer) [inline examples/misc/rasterization_demo.py:66]: that's not true, texts can be rasterized just fine nowadays.

File: lib/matplotlib/artist.py [hunk 3]
```diff
@@ -887,9 +887,15 @@ def get_rasterized(self):
 
     def set_rasterized(self, rasterized):
         """
-        Force rasterized (bitmap) drawing in vector backend output.
+        Force rasterized (bitmap) drawing for vector graphics output.
 
-        Defaults to None, which implies the backend's default behavior.
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply lib/matplotlib/artist.py:891]: I suspect that this wording means "in Agg we _always_rasterize, in svg the default is to emit vector data", i.e. the default value is to do what ever the backend does (at least that is how I have always read).

[the remark this answers is not in the document]
Comment (timhoffm) [reply lib/matplotlib/artist.py:891]: 👍 "backends default behavior" is still a bit weird wording. To me, "default" implies that one can configure it to something else.

Anyway, the wording is gone. And a possible removal of None is for another PR.

File: examples/misc/rasterization_demo.py [hunk 4]
```diff
@@ -1,20 +1,37 @@
...
+at the cost of a fixed resolution.
 
-Individual artists can be rasterized for saving to a vector backend
-such as PDF, SVG, or PS as embedded images.  This can be useful to
-reduce the file size of large artists, while maintaining the
+Whether rasterization should be used can be specified per artist.  This can be
+useful to reduce the file size of large artists, while maintaining the
 advantages of vector graphics for other artists such as the axes
-and annotations.  For instance a complicated `~.Axes.pcolormesh` or
+and texts.  For instance a complicated `~.Axes.pcolormesh` or
```
Comment (QuLogic) [inline examples/misc/rasterization_demo.py:13]:
```suggestion
and text.  For instance a complicated `~.Axes.pcolormesh` or
```

File: examples/misc/rasterization_demo.py [hunk 5]
```diff
@@ -1,20 +1,37 @@
...
+  less than the given value.
+
+The storage size and the resolution of the rasterized artist is determined by
+its physical size and the value of the ``dpi`` parameter passed to
+`~.Figure.savefig`.
+
+.. note::
+
+    The image of this example shown in the HTML documentation is not a vector
+    graphics. Therefore, it cannot illustrate the rasterization effect. Please
```
Comment (QuLogic) [inline examples/misc/rasterization_demo.py:32]:
```suggestion
    graphic. Therefore, it cannot illustrate the rasterization effect. Please
```

**Review by QuLogic — APPROVED** [review]

**Comment by timhoffm** [comment]
I've committed @QuLogic's changes via GitHub interface. Please squash-merge.

**Pull request merged**